docs/connect.html

```diff
@@ -543,6 +543,67 @@ <h6>ACL requirements</h6>
+    </ul>
+
+    <p>If the verification step succeeds, then your current set of installed plugins are compatible, and it should be safe to change the <code>plugin.discovery</code> configuration to <code>SERVICE_LOAD</code>. If you change the set of already-installed plugins, they may no longer be compatible, and you should repeat the above verification. If the verification fails, you must address the incompatible plugins before using the <code>SERVICE_LOAD</code> strategy.</p>
+
+    <h5><a id="connect_plugindiscovery_migrateartifact" href="#connect_plugindiscovery_migrateartifact">Operators: Artifact Migration</a></h5>
+
+    <p>As an operator of Connect, if you discover incompatible plugins, there are multiple ways to try to resolve the incompatibility. They are listed below from most to least preferable.</p>
+
+    <ol>
+        <li>Upgrade your incompatible plugins to the latest release version from your plugin provider.</li>
+        <li>Contact your plugin provider and request that they migrate the plugin to be compatible, following the <a href="#connect_plugindiscovery_migratesource">source migration instructions</a>, and then upgrade to the migrated version.</li>
+        <li>Migrate the plugin artifacts yourself using the included migration script.</li>
+    </ol>
+
+    <p>The migration script is located in <code>bin/connect-plugin-path.sh</code> and <code>bin\windows\connect-plugin-path.bat</code> of your Kafka installation. The script can migrate incompatible plugin artifacts already installed on your Connect worker's <code>plugin.path</code> by adding or modifying JAR or resource files. This is not suitable for environments using code-signing, as this may change the artifacts such that they will fail signature verification. View the built-in help with <code>--help</code>.</p>
+
+    <p>To perform a migration, first use the <code>list</code> subcommand to get an overview of the plugins available to the script. You must tell the script where to find plugins, which can be done with the repeatable <code>--worker-config</code>, <code>--plugin-path</code>, and <code>--plugin-location</code> arguments. The script will only migrate plugins present in the paths specified, so if you add plugins to your worker's classpath, then you will need to specify those plugins via one or more <code>--plugin-location</code> arguments.</p>
+
+    <p>Once you see that all incompatible plugins are included in the listing, you can proceed to dry-run the migration with <code>sync-manifests --dry-run</code>. This will perform all parts of the migration, except for writing the results of the migration to disk. Note that the <code>sync-manifests</code> command requires all specified paths to be writable, and may alter the contents of the directories. Make a backup of the specified paths, or copy them to a writable directory.</p>
```

Comment:
What does "Once you see that all incompatible plugins are included in the listing" mean? Does a user need to do more than just invoke the `connect-plugin-path list` command for this? If not, how about "If incompatible plugins are shown in the listing"?

Reply:
It covers the case where `--worker-config` doesn't cover everything. I was imagining this workflow:

1. Operator receives a list of incompatible plugins from an error message or warning in production
2. Operator gets a dev environment and runs `bin/connect-plugin-path.sh list --worker-config config/connect-distributed.properties` and sees a listing
3. Operator verifies that every plugin in the error listing is in the `list` output.
4. If a plugin is missing, the operator figures out where that is coming from (e.g. from the classpath)
5. Operator adds `--plugin-location` arguments and repeats step 3-5 until all plugins are present
6. Operator moves on to `sync-manifests --dry-run`.

If I change this to `If incompatible plugins are shown` then the sync-manifests may be incomplete, and the verification steps may fail. I guess since they should do that verification anyway, having them cross-check the errors and list output is a bit redundant. WDYT?

Reply:
Ahh, thanks for enlightening me! I had a different flow in mind where cluster administrators relied more-heavily on the `list` command to discover incompatible plugins. If we assume that Connect worker startup is the primary mechanism for users to discover incompatible plugins, then I think the current language is better than my suggestion.

> If I change this to `If incompatible plugins are shown` then the sync-manifests may be incomplete, and the verification steps may fail. I guess since they should do that verification anyway, having them cross-check the errors and list output is a bit redundant. WDYT?

I actually really like establishing this expectation. We could add 1-2 extra sentences instructing users to take note of the list of incompatible plugins reported by their Connect worker during startup. If we do that, then the language used here wouldn't really need any modifications since "incompatible plugins" would clearly reference that list.

Reply:
I added a "take note" mention to the verification section, and a "Be sure to compare" mention here as a call back. It also meshes well with the special case for the classpath plugins mentioned immediately before.

Reply:
LGTM, thanks!